Source/Urho3D/Scene/Serializable.h

```diff
@@ -268,6 +273,10 @@ namespace AttributeMetadata
 /// Define an enum attribute with custom setter and getter. Zero-based enum values are mapped to names through an array of C string pointers.
 #define URHO3D_CUSTOM_ENUM_ATTRIBUTE(name, getFunction, setFunction, enumNames, defaultValue, mode) context->RegisterAttribute<ClassName>(Urho3D::AttributeInfo( \
     Urho3D::VAR_INT, name, Urho3D::MakeVariantAttributeAccessor<ClassName>(getFunction, setFunction), enumNames, static_cast<int>(defaultValue), mode))
+/// Define an object attribute. Object must be SharedPtr<> of Serializable or it's subclass.
+#define URHO3D_OBJECT_ATTRIBUTE(name, variable, mode) context->RegisterAttribute<ClassName>(Urho3D::AttributeInfo( \
+    VAR_CUSTOM_STACK, name, URHO3D_MAKE_GET_SET_OBJECT_ATTRIBUTE_ACCESSOR(variable), nullptr, MakeCustomValue(SharedPtr<Serializable>()), mode))
+static_assert(sizeof(CustomVariantValueImpl<SharedPtr<Serializable>>) <= VARIANT_VALUE_SIZE, "SharedPtr<> does not fit into variant.");
```

Comment:
Meh. If you want to have some check, let's move this assert into `Variant.h` and check that at least one pointer can fit into Variant w/o allocation.
I.e.
`static_assert(sizeof(CustomVariantValueImpl<void*>) <= VARIANT_VALUE_SIZE, "Pointer does not fit into Variant without allocation.");`

You can even add type trait for this condition if you are not too lazy.

Reply:
Could also have `sizeof(CustomVariantValueImpl<SharedPtr<Serializable>>) <= VARIANT_VALUE_SIZE ? VAR_CUSTOM_STACK : VAR_CUSTOM_HEAP` in the macro and remove `static_assert()` altogether. It would still always evaluate to `VAR_CUSTOM_STACK` though. But in case something goes wrong it would still work even if it does heap allocation. How does that sound?